plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java

```diff
@@ -134,8 +148,9 @@ private List<BigQuerySplit> readFromBigQuery(ConnectorSession session, TableDefi
             // Storage API doesn't support reading wildcard tables
             return ImmutableList.of(BigQuerySplit.forViewStream(columns, filter));
         }
-        if (type == MATERIALIZED_VIEW || type == EXTERNAL) {
+        if (type == MATERIALIZED_VIEW || (type == EXTERNAL && connectionId.isEmpty())) {
             // Storage API doesn't support reading materialized views and external tables
+            // without connection. Storage API does support BigLake external tables.
```

Comment:
What happens if the connection uses non-BigLake as the source? e.g. MySQL
![Screenshot 2024-03-14 at 6 45 33](https://github.com/trinodb/trino/assets/6237050/29e42650-d998-47bd-8289-0e4894edad7e)

Reply:
You won't be able to create an external table with a non-BigLake connections such as Cloud SQL.

Reply:
Can you share the source? For instance, [Create object tables](https://cloud.google.com/bigquery/docs/object-tables) contains the example with connection. This is different page from BigLake tables.

Reply:
I'm not asking if storage APIs support BigLake tables or not. You said "You won't be able to create an external table with a non-BigLake connections". However, https://cloud.google.com/bigquery/docs/object-tables indicates that connection-id can be used for other tables.

Reply:
For BigLake tables, the connection type has to be BigLake (a.k.a. `CLOUD_RESOURCE`): that was your original question. Yes, the BigLake connection could also be used for creating object tables. I don't think object tables are particularly interesting from a Trino perspective: they are primarily used for unstructured data analysis using BQML.

Reply:
>  I don't think object tables are particularly interesting from a Trino perspective

My question is the behavior when accessing object tables with this change. It's not covered by tests in my understanding.

Reply:
Object tables are not supported by the storage read API because there is no use case for it currently. I will exclude object tables from this code path, so that it goes through the query path. This requires some [changes](https://github.com/googleapis/java-bigquery/issues/3216) in the BigQuery Java client library to expose the `objectMetadata` field and we will need to upgrade Trino to the newer client libraries. I will contribute that change and update this PR.

Reply:
@anoopj What is your estimation on when newer BigQuery client libraries will be ready to be included to Trino?

Reply:
It may take a couple of weeks.

Reply:
@anoopj do you have any update?

Reply:
My [PR](https://github.com/googleapis/java-bigquery/pull/3217) was merged last week. I'm waiting for the new release of the client libraries, which is scheduled to happen next week.

Reply:
Just a heads-up. I'm busy with other projects and won't be able to come back to this in the next few weeks.

Reply:
@anoopj could you please share your plans regarding this PR?

Reply:
@ssheikin Sorry for the delay. I am not planning to work on this in the near term. Did you need this feature?

Reply:
Thank you! Yeah, we were interested in it. Do you know if there is someone who is willing to take over this PR?